### PR TITLE
Add assert_true and assert_false

### PR DESCRIPTION
**Pull request by Chemaclass**
## 🔖 Changes

- Add `assert_true`
- Add `assert_false` 

## ✅ To-do list

- [x] I updated the `CHANGELOG.md` to reflect the new feature or fix
- [x] I updated the documentation to reflect the changes

Screenshot 2024-10-04 at 22 26 30

### CODE REVIEW

**Review by JesusValera — APPROVED** [review]

File: src/assert.sh [hunk 1]
```diff
@@ -9,6 +9,36 @@ function fail() {
...
+  fi
+
+  state::add_assertions_passed
+}
+
+function assert_false() {
+  local actual="$1"
+
+  if [[ "$actual" != "false" && "$actual" != "1" ]]; then
+    local label
```
Comment (skinner-m-c) [inline src/assert.sh:31]: [nit] Note that this does not seem to work on functions. If I pass `assert_true ls` the assertion fails. A potential problem may be that the concept of true and false as used in `if` statements does not apply with this assertion. If statements really just match the exit code of the expression. Bash makes use of the `test` program to perform much of the boolean logic in BASH, which has the `[[ ... ]]` shorthand.

Future work might want to test if the value passed is a function (`type -f <var>`) or a value and do something. different.

**Review by skinner-m-c — COMMENTED** [review]
Thanks for adding true and false assertions.

File: tests/unit/assert_test.sh [hunk 2]
```diff
@@ -10,6 +10,26 @@ function test_unsuccessful_fail() {
     "$(fail "Failure message")"
 }
 
+function test_successful_assert_true() {
+  assert_empty "$(assert_true true)"
```
[the remark this answers is not in the document]
Comment (Chemaclass) [reply tests/unit/assert_test.sh:14]: Would you want to do it? Otherwise, I will :)

File: src/assert.sh (hunk 1 above)
Comment (Chemaclass) [reply src/assert.sh:31]: I will work on this, many thanks!!

File: tests/unit/assert_test.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (skinner-m-c) [reply tests/unit/assert_test.sh:14]: I am fine with you doing it.

[the remark this answers is not in the document]
Comment (Chemaclass) [reply tests/unit/assert_test.sh:14]: done: https://github.com/TypedDevs/bashunit/commit/247ef1684532392e4da05ef476b37aff2a7fac3a

**Pull request merged**